hub/core/serialize.py

```diff
@@ -312,10 +313,17 @@ def serialize_input_samples(
         buff = bytearray()
         nbytes = []
         shapes = []
+        expected_dim = len(meta.max_shape)
         for sample in samples:
             byts, shape = _serialize_input_sample(
                 sample, sample_compression, dtype, htype
             )
+            if htype == "image" and config.CONVERT_GRAYSCALE_IMAGES_TO_3D:
```

Comment:
are we sure that we only want to do this for htype "image"? If the user doesn't specify the htype he'll still face the grayscale problem

Reply:
yeah, i feel like a constant here isn't ideal. this should be an actual htype parameter

Reply:
@AbhinavTuli: code changed to address comment.

@mccrearyd: constant has been removed.